x-pack/plugins/spaces/server/routes/api/v1/spaces.js

```diff
@@ -40,6 +45,41 @@ export function initSpacesApi(server) {
     }
   });
 
+  server.route({
+    method: 'GET',
+    path: '/api/spaces/v1/spaces/_active',
+    async handler(request, reply) {
+      const basePath = request.getBasePath();
+
+      const spaceContext = getSpaceUrlContext(basePath);
+
+      if (!spaceContext) {
+        return reply();
+      }
+
+      try {
+        const client = request.getSavedObjectsClient();
+
+        const {
+          saved_objects: spaces = []
+        } = await client.find({
+          type: 'space',
+          search: `"${spaceContext}"`,
+          search_fields: ['urlContext'],
+        });
+
+        if (spaces.length === 0) {
+          return reply();
+        }
+
+        return reply(convertSavedObjectToSpace(spaces[0]));
```

Comment:
Can we modify this to throw that Error that we talked about if we find more than one space that matches the url-context?

Reply:
What should our behavior be when we detect this? Should a banner display at the top of the screen, should we redirect to the default space, or something else?

I don't think we should allow the user to stay in this Space, because we don't _really_ know which space they intended to visit. If security wasn't an issue, then we could send them to the management screen, but not everybody will have access to it. We could also send them to a dedicated error page that explains what happened...maybe with a link to the management screen?

Similarly, what should the experience be when navigating to a space that doesn't exist?

Reply:
Most 500s from APIs will cause that toast notification that an error occurred, so doing the same thing here (if possible) makes sense to me. We do have a fatal error page that we could potentially use, but the toast notification is probably more consistent.